x/perp/keeper/clearing_house.go

```diff
@@ -205,6 +207,22 @@ func (k Keeper) afterPositionUpdate(
 	})
 }
 
+// UpdateMetrics recalculates perp metrics for a particular pair.
+func (k Keeper) UpdateMetrics(ctx sdk.Context, pair common.AssetPair) {
+	pairString := pair.String()
+	var netSize = sdk.NewDec(0)
+
+	iter := k.Positions.Iterate(ctx, collections.PairRange[common.AssetPair, sdk.AccAddress]{})
+	defer iter.Close()
+
+	for ; iter.Valid(); iter.Next() {
+		if pair == iter.Key().K1() {
+			netSize = netSize.Add(iter.Value().Size_)
+		}
+	}
+	k.Metrics.Insert(ctx, pairString, types.Metrics{Pair: pairString, NetSize: netSize})
```

Comment:
It will be a lot more scalable to compute this when swaps occur on the virtual pool. On calls of `SwapQuoteForBase` and `SwapBaseForQuote`, we can add a small hook/fn to update the global base amount.

Then in the end block hook, we can simply grab the `netSize` from the KV store.